File: front/src/app/Models/Serology.ts
import { SerologyType, SerologyInterpretation } from "./SerologyEnums";

export class Serology {
  id!: number;
  date!: Date;
  type!: typeof SerologyType[keyof typeof SerologyType];
  value!: number;
  interpretation!: typeof SerologyInterpretation[keyof typeof SerologyInterpretation];
  comment!: string;
  consultationId!: number;
}

File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/AnalyseDTOs/Serology.java
package org.example.medicalreport.Models.AnalyseDTOs;

import lombok.Data;

import java.time.LocalDate;
@Data
public class Serology {
    long id;
    LocalDate date;
    int type;
    double value;
    int interpretation;
    String comment;
    long consultationId;
}
